### PR TITLE
Grants form

### PR DESCRIPTION
**Pull request by bswags**
Pretty complex form that allows for multiple entries of 3 different funding types (venture, optimism grants, other)

### CODE REVIEW

**Review by tinaszheng — APPROVED** [review]
🫡very nice use of react form hooks

File: src/components/projects/grants/GrantsForm.tsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+      setOptimismFields([])
+      setOtherFields([])
+      setSelectedNone(!selectedNone)
+    }
+  }
+
+  const onAddFundingType = async (type: FundingType) => {
+    if (type === "venture") {
+      const valid = await form.trigger("venture")
+      console.log("Valid:", valid)
```
Comment (tinaszheng) [inline src/components/projects/grants/GrantsForm.tsx:123]: delet?

File: src/components/projects/grants/GrantsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+      } else {
+        setVentureFields([])
+      }
+    } else if (type === "optimism") {
+      if (optimismFields.length === 0) {
+        setSelectedNone(false)
+        addOptimismField({
+          type: "",
+          link: "",
+          amount: "" as unknown as number, // We coerce it with zod, so this is a better empty UI
```
Comment (tinaszheng) [inline src/components/projects/grants/GrantsForm.tsx:93]: can it do `undefined` as empty state instead?

Comment (bswags) [reply src/components/projects/grants/GrantsForm.tsx:93]: no bc it's a required field 😵‍💫

File: src/components/projects/grants/GrantsForm.tsx (hunk 1 above)
Comment (bswags) [reply src/components/projects/grants/GrantsForm.tsx:123]: omg

**Pull request merged**